src/Views/AuthorView.vala

```diff
@@ -23,7 +23,12 @@ private class AppCenter.AuthorView : Gtk.Box {
             return;
         }
 
-        var author_packages = AppCenterCore.FlatpakBackend.get_default ().get_packages_by_author (package.author, AUTHOR_OTHER_APPS_MAX);
+        var author_packages = AppCenterCore.FlatpakBackend.get_default ().get_packages_by_author_id (package.author_id, AUTHOR_OTHER_APPS_MAX);
+
+        if (author_packages.size <= 1) {
+            author_packages = AppCenterCore.FlatpakBackend.get_default ().get_packages_by_author (package.author, AUTHOR_OTHER_APPS_MAX);
+        }
+
```

Comment:
I guess both `package.author_id` and `package.component.get_developer ().get_id ()` would be null and it would match in the get_packages_by_author_id method if the developer id is not provided

```suggestion
        Gee.Collection<Package> author_packages;
        if (package.author_id != null) {
            author_packages = AppCenterCore.FlatpakBackend.get_default ().get_packages_by_author_id (package.author_id, AUTHOR_OTHER_APPS_MAX);
        } else {
            author_packages = AppCenterCore.FlatpakBackend.get_default ().get_packages_by_author (package.author, AUTHOR_OTHER_APPS_MAX);
        }

```

Reply:
Warnings shown when I navigate to an app page that does not provide developer id:

```
** (io.elementary.appcenter:5675): CRITICAL **: 22:39:07.544: app_center_core_flatpak_backend_get_packages_by_author_id: assertion 'author_id != NULL' failed

** (io.elementary.appcenter:5675): CRITICAL **: 22:39:07.545: gee_collection_get_size: assertion 'self != NULL' failed
```

Reply:
Can you give me an example of application which has no `author_id`?

Reply:
Apps that does not have the `<developer id="foo.bar">` tag. For example, [Pin It!](https://appcenter.elementary.io/com.github.ryonakano.pinit/) published on AppCenter is not updated recently and thus does not have it.

Reply:
Good! Those logs dissapeared with the last changes. :smile: